### PR TITLE
Package disml.0.2.4

### PR DESCRIPTION
**Pull request by Mishio595**
### `disml.0.2.4`
An OCaml library for interfacing with the Discord API
Dis.ml is a library that provides a high-level interface to the Discord API.
Key features include:
* Automatic sharding
* Deserialization of Discord objects to record types with related helper methods
* Automatic rate-limiting

For examples, see `/bin` in the git repo.

---
* Homepage: https://gitlab.com/Mishio595/disml
* Source repo: git+https://gitlab.com/Mishio595/disml
* Bug tracker: https://gitlab.com/Mishio595/disml/issues

---
:camel: Pull-request generated by opam-publish v2.0.0

### CODE REVIEW

File: packages/disml/disml.0.2.4/opam [hunk 1]
```diff
@@ -0,0 +1,44 @@
...
+Dis.ml is a library that provides a high-level interface to the Discord API.
+Key features include:
+* Automatic sharding
+* Deserialization of Discord objects to record types with related helper methods
+* Automatic rate-limiting
+
+For examples, see `/bin` in the git repo.
+"""
+depends: [
+  "ocaml" {>= "4.07.0"}
```
Comment (kit-ty-kate) [inline packages/disml/disml.0.2.4/opam]: is there a reason why you rely on such a recent version of OCaml?

File: packages/disml/disml.0.2.4/opam [hunk 2]
```diff
@@ -0,0 +1,44 @@
...
+  "ppx_deriving_yojson" {>= "3.1"}
+  "websocket-async" {>= "2.12"}
+  "decompress"
+  "odoc" {with-doc & >= "1.3.0"}
+  "yojson" {>= "1.6.0"}
+]
+build: [
+  ["dune" "subst"] {pinned}
+  ["dune" "build" "-p" name "-j" jobs]
+  ["dune" "build" "@doc" "-j" jobs] {with-doc}
```
Comment (kit-ty-kate) [inline packages/disml/disml.0.2.4/opam]: this line probably also needs `"-p" name` like the previous one

File: packages/disml/disml.0.2.4/opam [hunk 3]
```diff
@@ -0,0 +1,44 @@
...
+depends: [
+  "ocaml" {>= "4.07.0"}
+  "dune" {build & >= "1.3.0"}
+  "async_ssl" {>= "v0.11.0"}
+  "core" {>= "v0.11.3"}
+  "cohttp-async" {>= "1.2.0"}
+  "ppx_sexp_conv" {>= "v0.11.2"}
+  "ppx_deriving_yojson" {>= "3.1"}
+  "websocket-async" {>= "2.12"}
+  "decompress"
```
Comment (kit-ty-kate) [inline packages/disml/disml.0.2.4/opam]: is there no lower-bound constraints for decompress. The API changed a lot between versions and I'm worried it wouldn't compile with earlier versions. Have you tried earlier versions?

**Comment by kit-ty-kate** [comment]
something is wrong with the dependency constraints but I'm not sure why. Pinning the package directly from the repository seems to work locally but the CI says otherwise..

File: packages/disml/disml.0.2.4/opam (hunk 1 above)
Comment (Mishio595) [reply packages/disml/disml.0.2.4/opam]: I haven't tested with older compiler versions, testing how far back this is compatible currently.

File: packages/disml/disml.0.2.4/opam (hunk 2 above)
Comment (Mishio595) [reply packages/disml/disml.0.2.4/opam]: Wasn't aware that was needed, still fairly new to opam and dune. I'll add that in, thanks.

**Comment by Mishio595** [comment]
I think I see the problem. Due to the changes in `yojson.1.6.0` I had updated to using the `t` types instead of `json`, but the latest release of `ppx_deriving_yojson` does not yet have this merged. I had been locally pinned to the dev repo for the ppx.

I'll make some changes to require yojson < 1.6.0 so the ppx is happy until we get a new release from them.

**Comment by Mishio595** [comment]
Since this would require a new release on my end to revert, should I go ahead and close this PR and open a new one for the subsequent release?

**Pull request closed without merging**